egs/hub4_english/s5/local/prepare_dict.sh

```diff
@@ -0,0 +1,213 @@
+
+
+#(2) Dictionary preparation:
+
+
+if [ $stage -le 0 ]; then
+  # Make phones symbol-table (adding in silence and verbal and non-verbal noises at this point).
+  # We are adding suffixes _B, _E, _S for beginning, ending, and singleton phones.
+
+  # silence phones, one per line.
+  (echo SIL; echo SPN; echo NSN; echo UNK;) > $dir/silence_phones.txt
+  echo SIL > $dir/optional_silence.txt
+
+  # nonsilence phones; on each line is a list of phones that correspond
+  # really to the same base phone.
+  cat $dir/cmudict/cmudict.0.7a.symbols | perl -ane 's:\r::; print;' | \
+    perl -e 'while(<>){
+  chop; m:^([^\d]+)(\d*)$: || die "Bad phone $_";
+  $phones_of{$1} .= "$_ "; }
+  foreach $list (values %phones_of) {print $list . "\n"; } ' \
+    > $dir/nonsilence_phones.txt || exit 1;
+
+  # A few extra questions that will be added to those obtained by automatically clustering
+  # the "real" phones.  These ask about stress; there's also one for silence.
+  cat $dir/silence_phones.txt| awk '{printf("%s ", $1);} END{printf "\n";}' > $dir/extra_questions.txt || exit 1;
+  cat $dir/nonsilence_phones.txt | perl -e 'while(<>){ foreach $p (split(" ", $_)) {
+  $p =~ m:^([^\d]+)(\d*)$: || die "Bad phone $_"; $q{$2} .= "$p "; } } foreach $l (values %q) {print "$l\n";}' \
+    >> $dir/extra_questions.txt || exit 1;
+
+  grep -v ';;;' $dir/cmudict/cmudict.0.7a | \
+    perl -ane 'if(!m:^;;;:){ s:(\S+)\(\d+\) :$1 :; print; }' \
+    > $dir/dict.cmu || exit 1;
+
+  # Add to cmudict the silences, noises etc.
+
+  (echo '!SIL SIL'; echo '<SPOKEN_NOISE> SPN'; echo '<unk> UNK'; echo '<NOISE> NSN'; ) | \
+    cat - $dir/dict.cmu > $dir/lexicon2_raw.txt
+  awk '{print $1}' $dir/lexicon2_raw.txt > $dir/orig_wordlist
+
+  cat <<EOF >$dir/silence_phones.txt
+SIL
+SPN
+NSN
+UNK
+EOF
+
+fi
+
+
+if [ $stage -le 2 ]; then
+  if [ ! -f exp/g2p/.done ]; then
+    steps/dict/train_g2p.sh --cmd "$train_cmd" \
+      --silence-phones $dir/silence_phones.txt \
+      $dir/dict.cmu exp/g2p
+    touch exp/g2p/.done
+  fi
+fi
+
+export PATH=$PATH:`pwd`/local/dict
+
+if [ $stage -le 3 ]; then
+  cat $wordlist | python -c '
+import sys
+
+words = {}
+for line in open(sys.argv[1]).readlines():
+  words[line.strip()] = 1
+
+oovs = {}
+for line in sys.stdin.readlines():
+  word = line.strip()
+  if word not in words:
+    oovs[word] = 1
+
+for oov in oovs:
+  print (oov)' $dir/orig_wordlist | sort -u > $dir/oovlist
+
+  cat $dir/oovlist | \
+    get_acronym_prons.pl $dir/lexicon2_raw.txt > $dir/dict.acronyms
+fi
+
+mkdir -p $dir/f $dir/b # forward, backward directions of rules...
+
+if [ $stage -le 4 ]; then
+  # forward is normal suffix
+  # rules, backward is reversed (prefix rules).  These
+  # dirs contain stuff we create while making the rule-based
+  # extensions to the dictionary.
+
+  # Remove ; and , from words, if they are present; these
+  # might crash our scripts, as they are used as separators there.
+  filter_dict.pl $dir/dict.cmu > $dir/f/dict
```

Comment:
You seem to have inherited some old WSJ scripts for extending a dictionary.   I believe this works worse than g2p.py, when we tested.  Is this really needed?